cop.h

```diff
@@ -434,6 +434,15 @@ the octets.
 
 #include "mydtrace.h"
 
+/* keep in sync with feature.h (which will complain if this is out of sync)
+ */
+#define COP_FEATURE_SIZE 1
+
+/* make this a struct so we can copy the feature bits with assignment */
+struct cop_feature_t {
```

Comment:
I'd go with:
```
typedef U32 cop_feature_block_t;
typedef cop_feature_block_t cop_feature_t [COP_FEATURE_SIZE];
```

for macros:
```
#define FEATURE_A 0
...
#define FEATURE_B 126
#define FEATURE_MAX_127


#define COP_FEATURE_SIZE ((COP_FEATURE_MAX / sizeof (cop_feature_block_t))

#define bitsizeof(X) (8 * sizeof (X))
#define IS_FEATURE_ENABLED(Flags, Feature) \
   Flags[Feature / bitsizeof (cop_feature_block_t)] & (1 << (Feature % bitsizeof (cop_feature_block_t))
....

For constants operations are evaluated at compile time.
Macros can adapt to best int size, for example:
```
typedef
#if COP_FEATURE_MAX > bitsize (U32) && U64_BITOP_IS_FASTER_THAN_U32
  U64
#else
  U32
#endif
cop_feature_block_t;